Core/src/RequestWrapper.php

```diff
@@ -305,48 +307,58 @@ private function applyHeaders(RequestInterface $request, array $options = [])
             unset($options['retryHeaders']);
         }
 
+        $request = Utils::modifyRequest($request, ['set_headers' => $headers]);
+
         if ($this->shouldSignRequest) {
             $quotaProject = $this->quotaProject;
-            $token = null;
 
             if ($this->accessToken) {
-                $token = $this->accessToken;
+                $request = $request->withHeader('authorization', 'Bearer ' . $this->accessToken);
             } else {
                 $credentialsFetcher = $this->getCredentialsFetcher();
-                $token = $this->fetchCredentials($credentialsFetcher)['access_token'];
+                $request = $this->addAuthHeaders($request, $credentialsFetcher);
 
                 if ($credentialsFetcher instanceof GetQuotaProjectInterface) {
                     $quotaProject = $credentialsFetcher->getQuotaProject();
                 }
             }
 
-            $headers['Authorization'] = 'Bearer ' . $token;
-
             if ($quotaProject) {
-                $headers['X-Goog-User-Project'] = [$quotaProject];
+                $request = $request->withHeader('X-Goog-User-Project', $quotaProject);
             }
         }
 
-        return Utils::modifyRequest($request, ['set_headers' => $headers]);
+        return $request;
     }
 
     /**
-     * Fetches credentials.
+     * Adds auth headers to the request.
      *
-     * @param FetchAuthTokenInterface $credentialsFetcher
+     * @param RequestInterface $request
+     * @param FetchAuthTokenInterface $fetcher
      * @return array
      * @throws ServiceException
      */
-    private function fetchCredentials(FetchAuthTokenInterface $credentialsFetcher)
+    private function addAuthHeaders(RequestInterface $request, FetchAuthTokenInterface $fetcher)
     {
         $backoff = new ExponentialBackoff($this->retries, $this->getRetryFunction());
 
         try {
             return $backoff->execute(
-                function () use ($credentialsFetcher) {
-                    if ($token = $credentialsFetcher->fetchAuthToken($this->authHttpHandler)) {
-                        return $token;
+                function () use ($request, $fetcher) {
+                    if (!$fetcher instanceof UpdateMetadataInterface ||
+                         ($fetcher instanceof FetchAuthTokenCache &&
+                            !$fetcher->getFetcher() instanceof UpdateMetadataInterface
+                         )
+                    ) {
+                        if ($token = $fetcher->fetchAuthToken()) {
```

Comment:
Is there a reason we aren't passing the `authHttpHandler` into this one anymore?
```suggestion
                    ) {
                        if ($token = $fetcher->fetchAuthToken($this->authHttpHandler)) {
```

Reply:
Whooops! How did I miss this! : '(.

Thanks a lot Brent on catching this!!

I've updated it. This was not flagged in any of the unit tests since it's a trivial task to pass the handlers and yet I missed that trivial part : (. Do you think we should have a test just to see if the supplied `authHttpHandler` is getting passed or not? Though this seems an overkill too.

Reply:
I think a test to check that the authHttpHandler is passed should be pretty easy, and also very important:

```php
$authHttpHandler = function (RequestInterface $request) {};
$fetcher = $this->prophesize(FetchAuthTokenInterface::class);
$fetcher->fetchAuthToken($authHttpHandler)
    ->shouldBeCalledOnce()
    ->willReturn(['access_token' => 'xyz');
$wrapper = new RequestWrapper([
    'authHttpHandler' => $authHttpHandler->reveal(),
    'credentialsFetcher' => $fetcher->reveal(),
]);
$wrapper->send($this->prophesize(RequestInterface::class)->reveal());
```

Reply:
Fixed in https://github.com/googleapis/google-cloud-php/pull/6738/commits/a7f696e2f489b3000eb6748665a3f1db7e43c6be